### PR TITLE
Ordinal block processing in SE

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
`Squash ImproveChecks` commit doesn't need a review

### CODE REVIEW

File: analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/CollectorTestCheck.cs [hunk 1]
```diff
@@ -30,20 +31,42 @@ namespace SonarAnalyzer.UnitTest.TestFramework.SymbolicExecution
     internal class CollectorTestCheck : SymbolicCheck
```
Comment (csaba-sagi-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/SymbolicExecution/CollectorTestCheck.cs:31]: Very nice idea, it simplifies the tests a lot.

**Review by csaba-sagi-sonarsource — APPROVED** [review]
Please improve the coverage. 100% should be easy to achieve.

**Comment by pavel-mikula-sonarsource** [comment]
> Please improve the coverage. 100% should be easy to achieve.

I had to remove the null check, because `null` can not pass through the public constructors due to NREs there.

**Comment by pavel-mikula-sonarsource** [comment]
cc @costin-zaharia-sonarsource

**Pull request merged**